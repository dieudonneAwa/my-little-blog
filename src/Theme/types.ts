export type Colors = {
  emerald: string;
  spacecadet: string;
  cultured: string;
  lightgrey: string;
  white: string;
  blueviolet: string;
  richblack: string;
};

export type Space = {
  none: string;
  nano: string;
  micro: string;
  mini: string;
  tiny: string;
  small: string;
  medium: string;
  large: string;
  big: string;
  huge: string;
  massive: string;
};

export type Font = {
  family: {
    poppins: string;
  };
  size: Space;
};

export type Radii = {
  none: string;
  small: string;
  medium: string;
  large: string;
  huge: string;
};

export type Article = {
  id: string;
  createdAt: string;
  title: string;
  content: string;
  createdBy: string;
};

export type Action = {
  type: string;
  payload?: any;
};

export type User = {
  id: string;
  name: string;
  avatar: string;
  createdAt: string;
};

export type Comment = {
  id: string;
  name: string;
  createdAt: string;
  content: string;
};

export type AppTheme = 'light' | 'dark';
