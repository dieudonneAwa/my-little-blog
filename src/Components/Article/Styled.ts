import styled from 'styled-components';
import { position, PositionProps } from 'styled-system';

export const ArticleHeading = styled.h1`
  font-size: ${(p) => p.theme.font.size.big};
  font-weight: 500;
  line-height: 1.3;
  margin-bottom: 1rem;

  @media (max-width: 414px) {
    font-size: ${(p) => p.theme.font.size.medium};
    font-weight: 600;
  }
`;

export const BackButton = styled.span<PositionProps>`
  font-size: ${(p) => p.theme.font.size.small};
  cursor: pointer;
  ${position}

  &:hover {
    color: ${(p) => p.theme.colors.emerald};
    svg path {
      fill: ${(p) => p.theme.colors.emerald};
    }
  }
`;

export const CommentName = styled.strong`
  font-size: ${(p) => p.theme.font.size.small};
  font-weight: 500;
`;
export const CommentTime = styled.p`
  font-size: calc(${(p) => p.theme.font.size.small} - 0.2rem);
  font-weight: 300;
  margin-top: -0.5rem;
`;
