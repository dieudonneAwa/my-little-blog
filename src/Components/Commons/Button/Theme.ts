import { css } from "styled-components";
import { darken, lighten, transparentize } from "polished";
import { font, space } from "../../../Theme/typography";

// BASE
const buttonBase = css`
  font-weight: 500;
  transition: all 0.2s linear;
  min-width: fit-content;
  font-size: ${font.size.medium};

  &:hover {
    cursor: pointer;
    transform: scale(1);
    transition: all 0.5s cubic-bezier(0.075, 0.82, 0.165, 1);
  }
  &:not(:hover) {
    transform: scale(1);
    transition: all 0.6s cubic-bezier(0.075, 0.82, 0.165, 1);
  }
  @media (max-width: 767px) {
    padding: ${space.small};
  }
`;

// COLOR

const emerald = css`
  background: ${(p) => p.theme.colors.emerald};
  color: ${(p) => p.theme.colors.white};
  box-shadow: -1px -1px 8px ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.emerald))};
  border: 1px solid ${(p) => p.theme.colors.emerald};

  &:hover {
    background: ${(p) => lighten(0.05, p.theme.colors.emerald)};
    box-shadow: 0px 0px 12px
      ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.emerald))};
  }
`;

const spacecadet = css`
  border: 1px solid ${(p) => p.theme.colors.spacecadet};
  background: ${(p) => p.theme.colors.spacecadet};
  color: ${(p) => p.theme.colors.white};

  &:hover {
    background: ${(p) => darken(0.05, p.theme.colors.spacecadet)};
    box-shadow: 0px 0px 12px
      ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.spacecadet))};
  }
`;

const blueviolet = css`
  border: 1px solid ${(p) => p.theme.colors.blueviolet};
  background: ${(p) => p.theme.colors.blueviolet};
  color: ${(p) => p.theme.colors.white};

  &:hover {
    background: ${(p) => darken(0.05, p.theme.colors.blueviolet)};
    box-shadow: 0px 0px 12px
      ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.blueviolet))};
  }
`;

const defaultcolor = css``;

const buttonColor: any = {
  emerald,
  spacecadet,
  default: defaultcolor,
  blueviolet,
};

// SIZE

const small = css`
  font-size: ${font.size.medium};
  padding: ${space.tiny} ${space.medium};
  border-radius: ${(p) => p.theme.radii.small};
`;

const medium = css`
  font-size: ${font.size.medium};
  padding: ${space.small} ${space.large};
  border-radius: ${(p) => p.theme.radii.small};
`;

const large = css`
  font-size: ${font.size.large};
  padding: ${space.medium} ${space.large};
  border-radius: ${(p) => p.theme.radii.medium};
`;

const buttonSize: any = {
  small,
  medium,
  large,
};

// SHAPE
const rounded = css``;

const square = css`
  border-radius: 0;
`;
const pill = css`
  border-radius: 50px;
`;

const buttonShape: any = {
  rounded,
  square,
  pill,
};

export { buttonBase, buttonColor, buttonSize, buttonShape };
