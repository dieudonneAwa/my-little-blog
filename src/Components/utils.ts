import { ShallowWrapper } from 'enzyme';

export const findByTestAttr = (wrapper: ShallowWrapper<void>, val: string) =>
  wrapper.find(`[data-test="${val}"]`);
