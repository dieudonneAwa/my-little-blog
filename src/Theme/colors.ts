const colors = {
  emerald: "#1EAE68",
  spacecadet: "#1b2546",
  cultured: "#F1F1F1",
  lightgrey: "#FBFBFB",
  white: "white",
  blueviolet: "#5621ab",
};

export default colors;
