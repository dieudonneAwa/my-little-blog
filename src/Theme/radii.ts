const radii = {
  none: '0',
  small: '3px',
  medium: '6px',
  large: '9px',
  huge: '50%',
}

export default radii
