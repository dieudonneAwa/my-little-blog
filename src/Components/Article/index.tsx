/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import Avatar from 'react-avatar';
import { Box } from '../Commons/Box';
import ArrowBackIcon from '../Commons/Icons/ArrowBack';
import { ArticleHeading, BackButton, CommentTime, CommentName } from './Styled';
import { Article, Comment } from '../../Theme/types';
import { AxiosResponse } from 'axios';
import Spinner from '../Commons/Spinner';

export type Props = {
  article: Article;
  fetchComments: () => Promise<AxiosResponse<any>>;
  comments: Comment[];
  loadingComments: boolean;
};

const ArticlePage = ({ article, fetchComments, comments, loadingComments }: Props) => {
  const router = useHistory();

  useEffect(() => {
    if (!comments.length) {
      fetchComments();
    }
  }, []);

  const handleClick = () => {
    router?.push('/');
  };

  return (
    <Box display="flex" flexDirection="column" alignItems="center" px={['1rem', '1rem']}>
      <BackButton
        position="absolute"
        top="3rem"
        left={['1rem', '1rem', '3rem']}
        data-test="go-back-btn"
        onClick={handleClick}
      >
        <ArrowBackIcon /> Back
      </BackButton>
      <Box width={['100%', '100%', '700px']} minHeight="250px">
        <ArticleHeading>{article?.title}</ArticleHeading>
        <p>{article?.content}</p>
      </Box>
      <Box
        width={['100%', '100%', '650px']}
        display="flex"
        flexDirection="column"
        alignItems="center"
        flexWrap={['wrap', 'nowrap']}
      >
        {loadingComments && (
          <Box display="flex" justifyContent="center" alignItems="center" height="300px">
            <Spinner />
          </Box>
        )}
        {comments?.map((comment) => (
          <Box
            borderTopWidth="1px"
            width="100%"
            minHeight="130px"
            borderStyle="solid"
            p="1rem"
            borderColor="cultured"
          >
            <Box display="flex" alignItems="center" flexWrap="nowrap">
              <Avatar name={comment.name} round size="35px" />
              <Box ml="0.5rem" p={0}>
                <CommentName>{comment.name}</CommentName> <br />
                <CommentTime>{new Date(comment.createdAt).toDateString()}</CommentTime>
              </Box>
            </Box>
            <p>{comment.content}</p>
          </Box>
        ))}
      </Box>
    </Box>
  );
};

export default ArticlePage;
