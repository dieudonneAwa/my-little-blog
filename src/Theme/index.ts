import { colors, darkThemeColors } from './colors';
import { Colors, Font, Radii, Space } from './types';
import { font, space } from './typography';
import radii from './radii';

export interface DefaultTheme {
  colors: Colors;
  font: Font;
  radii: Radii;
  space: Space;
}

const theme: DefaultTheme = {
  colors,
  font,
  radii,
  space,
};

const darkTheme: DefaultTheme = {
  ...theme,
  colors: darkThemeColors,
};

export { theme, darkTheme };
