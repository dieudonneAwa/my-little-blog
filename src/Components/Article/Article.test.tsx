/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Article, { Props } from './index';
import { findByTestAttr } from '../utils';
import { Article as IArticle } from '../../Theme/types';

const props: Props = {
  article: {} as IArticle,
  fetchComments: jest.fn(),
  comments: [
    {
      id: '1',
      name: 'name',
      content: 'Content',
      createdAt: new Date().toISOString(),
    },
  ],
  loadingComments: false,
};

describe('Article', (): void => {
  let wrapper: ShallowWrapper<void>;
  beforeEach((): void => {
    wrapper = shallow(<Article {...props} />);
  });

  it('should render Article component', (): void => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate back button click', (): void => {
    const backButton = findByTestAttr(wrapper, 'go-back-btn');
    backButton.simulate('click');
  });
});
