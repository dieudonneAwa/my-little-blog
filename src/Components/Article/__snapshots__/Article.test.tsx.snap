// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render Article component 1`] = `
<styled.div
  alignItems="center"
  display="flex"
  flexDirection="column"
  px={
    Array [
      "1rem",
      "1rem",
    ]
  }
>
  <styled.span
    data-test="go-back-btn"
    left={
      Array [
        "1rem",
        "1rem",
        "3rem",
      ]
    }
    onClick={[Function]}
    position="absolute"
    top="3rem"
  >
    <ArrowBackIcon />
     Back
  </styled.span>
  <styled.div
    minHeight="300px"
    width={
      Array [
        "100%",
        "100%",
        "700px",
      ]
    }
  >
    <styled.h1 />
    <p />
  </styled.div>
  <styled.div
    alignItems="center"
    display="flex"
    flexDirection="column"
    flexWrap={
      Array [
        "wrap",
        "nowrap",
      ]
    }
    width={
      Array [
        "100%",
        "100%",
        "500px",
      ]
    }
  >
    <styled.div
      borderColor="cultured"
      borderStyle="solid"
      borderTopWidth="1px"
      minHeight="130px"
      p="1rem"
      width="100%"
    >
      <styled.div
        alignItems="center"
        display="flex"
        flexWrap="nowrap"
      >
        <ForwardRef(withAvatarConfig)
          name="name"
          round={true}
          size="35px"
        />
        <styled.div
          ml="0.5rem"
          p={0}
        >
          <styled.strong>
            name
          </styled.strong>
           
          <br />
          <styled.p>
            Tue Apr 27 2021
          </styled.p>
        </styled.div>
      </styled.div>
      <p>
        Content
      </p>
    </styled.div>
  </styled.div>
</styled.div>
`;
